tests/test_iam.py

```diff
@@ -766,6 +766,120 @@ def test_iam_user_access_key_multi(self):
+
+    def test_iam_user_ssh_key_age_filter(self):
+        factory = self.replay_flight_data('test_iam_user_ssh_key_age_filter')
+        p = self.load_policy({
+            'name': 'iam-user-old-ssh-keys',
+            'resource': 'iam-user',
+            'filters': [
+                {'type': 'ssh-key', 'key': 'Status', 'value': 'Active'},
+                {'type': 'ssh-key', 'key': 'UploadDate',
+                 'value_type': 'age', 'value': 90, 'op': 'greater-than'},
+            ]},
+            session_factory=factory)
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+        self.assertEqual(len(resources[0]['c7n:matched-ssh-keys']), 1)
+
+    def test_iam_user_disable_ssh_keys(self):
+        factory = self.replay_flight_data('test_iam_user_disable_ssh_keys')
+        user_name = 'test2'
+        p = self.load_policy({
+            'name': 'iam-user-ssh-keys',
+            'resource': 'iam-user',
+            'filters': [
+                {'UserName': user_name},
+                {'type': 'ssh-key', 'key': 'Status', 'value': 'Active'},
+            ],
+            'actions': [
+                {'type': 'delete-ssh-keys', 'disable': True},
+            ]},
+            session_factory=factory)
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+        self.assertEqual(len(resources[0]['c7n:matched-ssh-keys']), 2)
+        client = p.session_factory().client('iam')
+        keys = client.list_ssh_public_keys(UserName=user_name)['SSHPublicKeys']
+        self.assertTrue(all(
+            key['Status'] == 'Inactive' for key in keys
+        ))
+
+    def test_iam_user_delete_ssh_keys(self):
+        factory = self.replay_flight_data('test_iam_user_delete_ssh_keys')
+        user_name = 'test2'
+        p = self.load_policy({
+            'name': 'iam-user-delete-ssh-keys',
+            'resource': 'iam-user',
+            'filters': [
+                {'UserName': user_name},
+                {'type': 'ssh-key', 'key': 'Status', 'value': 'Active'},
+            ],
+            'actions': [
+                {'type': 'delete-ssh-keys'},
+            ]},
+            session_factory=factory)
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+        self.assertEqual(len(resources[0]['c7n:matched-ssh-keys']), 2)
+        client = p.session_factory().client('iam')
+        keys = client.list_ssh_public_keys(UserName=user_name)['SSHPublicKeys']
+        self.assertEqual(len(keys), 0)
+
+    def test_iam_user_delete_old_ssh_keys(self):
+        factory = self.replay_flight_data('test_iam_user_delete_old_ssh_keys')
+        user_name = 'test2'
+        p = self.load_policy({
+            'name': 'iam-user-delete-old-ssh-keys',
+            'resource': 'iam-user',
+            'filters': [
+                {'UserName': user_name},
+            ],
+            'actions': [
+                {'type': 'delete-ssh-keys', 'age': 0},
```

Comment:
what's the purpose of age: 0, implicit filtering in actions isn't generally something we do, there's some specifics around access keys i know we implemented, just want to clarify. also could use a source: config

Reply:
hmm i was trying to follow the model from deleting access keys, but i'd rather handle filtering from an actual filter anyway. more dry run friendly, etc. i'll yank the age piece out.